pkg/install/install_test.go

```diff
@@ -439,7 +456,154 @@ func TestInstall(t *testing.T) {
 				returnedPath, err = installer.Install(tt.args.templatePath, tt.args.targetDir, tt.args.force)
 			}
 
-			if tt.expected.errorMsg != "" && err != nil {
+			if tt.expected.errorMsg != "" {
+				assert.Contains(t, err.Error(), tt.expected.errorMsg)
+			} else {
+				assert.NoError(t, err)
+			}
+			assert.Equal(t, tt.expected.filepath, returnedPath)
+		})
+	}
+}
+
+func TestInstaller_InstallFromConfig(t *testing.T) {
+	type args struct {
+		configFile string
+		targetDir  string
+		force      bool
+	}
+
+	type InstallFromConfigTest struct {
+		name              string
+		args              args
+		expected          expected
+		mocks             mocks
+		mockInstallConfig mockInstallConfig
+	}
+
+	extractionPath := "path/to/extract/to"
+
+	tests := []InstallFromConfigTest{
+		{
+			name: "Installs successfully and returns the correct namespaced path",
+			mockInstallConfig: mockInstallConfig{
+				metadata: config_processor.ConfigMetadata{
+					Author:  "puppetlabs",
+					Id:      "good-project",
+					Version: "0.1.0",
+				},
+				expectedConfigFile: filepath.Join(extractionPath, "good-project", "pct-config.yml"),
+			},
+			args: args{configFile: filepath.Join(extractionPath, "pct-config.yml"), targetDir: extractionPath},
+			expected: expected{
+				filepath: filepath.Join(extractionPath, "puppetlabs/good-project/0.1.0"),
+			},
+			mocks: mocks{
+				dirs: []string{
+					extractionPath,
+				},
+			},
+		},
+		{
+			name: "Install fails because of a missing attribute",
+			mockInstallConfig: mockInstallConfig{
+				metadata: config_processor.ConfigMetadata{
+					Author:  "puppetlabs",
+					Version: "0.1.0",
+				},
+				expectedConfigFile: filepath.Join(extractionPath, "good-project", "pct-config.yml"),
+				ErrResponse:        fmt.Errorf("attribute missing in config metadata: id"),
+			},
+			args: args{configFile: filepath.Join(extractionPath, "pct-config.yml"), targetDir: extractionPath},
+			expected: expected{
+				errorMsg: "attribute missing in config metadata: id",
+			},
+			mocks: mocks{
+				dirs: []string{
+					extractionPath,
+				},
+			},
+		},
+		// Neither of these tests work. Most likely a problem with the AFS.Rename(path1, path2) function (pkg/install/install.go:202)
```

Comment:
Whats the plan for this block here? Do we need to fix the tests or remove it?

Reply:
I believe the problem with these tests is to do with issues in the Afero module. I was going to keep them commented out so they could be run at a later date, or we can just remove them now.

Reply:
Yeah that sounds reasonable.. can we create an issue to track it please and update the comment with the issue number?